exercises/crypto-square/test/test_crypto_square.c

```diff
@@ -0,0 +1,106 @@
+#include "vendor/unity.h"
+#include "../src/crypto_square.h"
+#include <stdlib.h>
+
+void setUp(void)
+{
+}
+
+void tearDown(void)
+{
+}
+
+void test_empty_text_res_in_an_empty_ciphertext(void)
+{
+   const char *input = "";
+   const char *expected = "";
+   char *res = ciphertext(input);
+   TEST_ASSERT_NOT_NULL(res);
```

Comment:
You don't need these NULL checks by the way. `TEST_ASSERT_EQUAL_STRING` checks for that with a pretty good error message, and `free(NULL)` is a valid call.

The error message:
```
test/test_ciphertext.c:19:test_empty_text_res_in_an_empty_ciphertext:FAIL: Expected '' Was NULL
```

Reply:
I never checked the implementation of ```TEST_ASSERT_EQUAL_STRING``` just followed my own assumptions.

Nice catch - thanks!